tests/acceptance/default-layout-test.js

```diff
@@ -0,0 +1,56 @@
+import { test } from 'qunit';
+import Component from '@ember/component';
+import Controller from '@ember/controller';
+import hbs from 'htmlbars-inline-precompile';
+import moduleForAcceptance from '../../tests/helpers/module-for-acceptance';
+
+moduleForAcceptance('Acceptance | Default Layout Behavior');
+
+test('Assigning defaultLayout to a component should set it up as a layout if no layout was found [DEPRECATED]', function(assert) {
+  assert.expect(2);
+
+  this.application.register('template:defaultlayout',
+    hbs`<div id='wrapper'>{{#my-component}}{{text}}{{/my-component}}</div>`
+  );
+  this.application.register('controller:defaultlayout', Controller.extend({
+    text: 'outer'
+  }));
+  this.application.register('component:my-component', Component.extend({
+    text: 'inner',
+    defaultLayout: hbs`{{text}}-{{yield}}`
+  }));
+
+  visit('/defaultlayout');
+
+  andThen(function() {
+    assert.expectDeprecation();
```

Comment:
I would have assumed the deprecation would be fired sometime before here...

Reply:
@rwjblue It is. However the expectDeprecation is not catching it for some reason unless placed here. Not sure what is going on in either the background of the acceptance tests or with expectDeprecation to cause this. This issue (https://github.com/workmanw/ember-qunit-assert-helpers/issues/18) is what made me think about moving it to inside of the andThen.